store/store.go

```diff
@@ -531,11 +531,37 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
-// Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+type metadataInput struct {
+	SnapID   string `json:"snap_id"`
+	Channel  string `json:"channel"`
+	Revision int    `json:"revision"`
+	// FIXME: not in snapinfo yet
+	//Confinement string `json:"confinement"`
+}
+
+type metadataWrapper struct {
+	Snaps  []metadataInput `json:"snaps"`
+	Fields []string        `json:"fields"`
+}
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+// Updates returns the available updates for a list of snap identified by fullname with channel.
+func (s *SnapUbuntuStoreRepository) Updates(installed []*snap.Info, auther Authenticator) (snaps []*snap.Info, err error) {
+
+	// build input for the updates endpoint
+	is := make([]metadataInput, len(installed))
+	for i, inst := range installed {
+		is[i] = metadataInput{
+			SnapID:   inst.SnapID,
+			Channel:  inst.Channel,
```

Comment:
we have a Channel both on SnapState and the single SideInfos, not sure with rollbacks they are always be in sync, wondering which to pick and wondering whether we shouldn't pass snap.Info but only the precise required data